csharp/src/Ice/Reference.cs

```diff
@@ -1285,7 +1320,7 @@ internal void Write(OutputStream ostr)
                 var proxyData = new ProxyData20(Identity,
                                                 Protocol == Protocol.Ice2 ? null : Protocol,
                                                 Encoding == Encoding.V20 ? null : Encoding,
-                                                AdapterId.Length > 0 ? new string[] { AdapterId } : null,
+                                                Location.Count > 0 ? Location.ToArray() : null,
```

Comment:
Could the location data member of ProxyData20 be a read only list?

Reply:
Yes, but not with the current Slice to C# struct mapping.